src/components/AssetsPage/index.jsx

```diff
@@ -10,43 +10,98 @@ import { getAssets } from '../../data/actions/assets';
 import edxBootstrap from '../../SFE.scss';
 import styles from './AssetsPage.scss';
 
-class AssetsPage extends React.Component {
+export class AssetsPage extends React.Component {
   constructor(props) {
     super(props);
+
+    const hasAssets = this.hasAssets(props.assetsList);
+
     this.state = {
       checked: false,
+      hasAssets,
     };
   }
 
   componentDidMount() {
     this.props.getAssets(this.props.request, this.props.courseDetails);
   }
 
-  componentDidUpdate() {
-    this.props.getAssets(this.props.request, this.props.courseDetails);
+  componentWillReceiveProps(nextProps) {
+    if (this.state.hasAssets && this.hasNoAssets(nextProps.assetsList)) {
+      this.setState({
+        hasAssets: false,
+      });
+    } else if (!this.state.hasAssets && this.hasAssets(nextProps.assetsList)) {
+      this.setState({
+        hasAssets: true,
+      });
+    }
+  }
+
+  componentDidUpdate(prevProps) {
+    if (JSON.stringify(prevProps.request) !== JSON.stringify(this.props.request)) {
+      this.props.getAssets(this.props.request, this.props.courseDetails);
+    }
   }
 
+  hasNoAssets = assetsList => (
+    assetsList.length === 0
+  )
+
+  hasAssets = assetsList => (
+    assetsList.length > 0
+  )
+
   handleCheckBoxChange = (checked) => {
     this.setState({
       checked,
     });
   }
 
+  renderAssetsDropZone = () => (
+    <WrappedAssetsDropZone
+      maxFileSizeMB={this.props.uploadSettings.max_file_size_in_mbs}
+    />
+  )
+
+  renderAssetsPage = () => (
+    <React.Fragment>
```

Comment:
This is some new React 16.2 functionality. It basically allows us to return adjacent JSX elements without wrapping them in, say, a div or a span.

You can read more [here](https://reactjs.org/blog/2017/11/28/react-v16.2.0-fragment-support.html).